packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -734,3 +714,76 @@ async function areFoldersEmpty(folderPaths: string[]): Promise<boolean> {
 async function waitSeconds(second: number) {
   return new Promise((resolve) => setTimeout(resolve, second * 1000));
 }
+
+async function getDeploymentError(
+  deployCtx: DeployContext,
+  resourceGroupName: string,
+  deploymentName: string
+): Promise<any> {
+  let deployment;
+  try {
+    deployment = await deployCtx.client.deployments.get(resourceGroupName, deploymentName);
+  } catch (error) {
+    return undefined;
+  }
+  if (!deployment.properties?.error) {
+    return undefined;
+  }
+  const deploymentError: any = {
+    error: deployment.properties?.error,
+  };
+  const operations = await deployCtx.client.deploymentOperations.list(
+    resourceGroupName,
+    deploymentName
+  );
+  for (const operation of operations) {
+    if (operation.properties?.statusMessage?.error) {
+      if (!deploymentError.subErrors) {
+        deploymentError.subErrors = {};
+      }
+      const name = operation.properties.targetResource?.resourceName ?? operation.id;
+      deploymentError.subErrors[name!] = {
+        error: operation.properties.statusMessage.error,
+      };
+      if (
+        operation.properties.targetResource?.resourceType ===
+          ConstantString.DeploymentResourceType &&
+        operation.properties.targetResource?.resourceName &&
+        operation.properties.targetResource?.id
+      ) {
+        const resourceGroupName: string = getResourceGroupFromDeploymentId(
+          operation.properties.targetResource.id
+        );
+        const subError = await getDeploymentError(
+          deployCtx,
+          resourceGroupName,
+          operation.properties.targetResource?.resourceName
+        );
+        if (subError) {
+          deploymentError.subErrors[name!].inner = subError;
+        }
+      }
+    }
+  }
+  return deploymentError;
+}
+
+function getResourceGroupFromDeploymentId(deploymentId: string | undefined) {
+  if (deploymentId) {
+    const resultArr = deploymentId.match(/\/resourceGroups\/[a-zA-Z_0-9-]*/);
+    if (resultArr && resultArr.length > 0) {
+      return resultArr[0].substring(16);
+    }
+  }
+  return "";
+}
+
+function buildDeploymentErrorMessage(failedDeployments: string[]): Result<void, FxError> {
+  const format = failedDeployments.map((deployment) => deployment + " module");
```

Comment:
better naming? maybe formattedDeploymentName or something else

Reply:
It is fixed. Thanks.